### PR TITLE
Python 3 in environment if both python 2 and python 3 are installed

### PR DESCRIPTION
**Pull request by jack17529**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

This fixes issue https://github.com/coala/coala/issues/3921

### CODE REVIEW

**Comment by corona10** [comment]
Please test on your local environment.
- `coala`
- `py.test`
will help you. 😄

**Comment by NiklasMM** [comment]
This should only be one commit, but there are 4 in this PR. Please rebase your branch, so it only has one commit. (See https://coala.io/rebase for help)

Also you must adress the issues pointed out by gitmate-bot. Run coala locally as @corona10 suggested before you push.

**Comment by jack17529** [comment]
@corona10 this is the error I am getting when I run py.test
FAIL Required test coverage of 100% not reached. Total coverage: 99.81%
=========================== short test summary info ============================
SKIP [1] tests/results/DiffTest.py:208: /home/jack/env/bin/python3: undefined symbol: clang_annotateTokens. Please ensure that your python bindings are compatible with your libclang.so version.

============================ pytest-warning summary ============================
WC1 /home/jack/coala/tests/results/DiffTest.py cannot collect test class 'SkipTest' because it has a __init__ constructor
========== 612 passed, 1 skipped, 1 pytest-warnings in 32.04 seconds ===========

**Comment by jack17529** [comment]
@corona10 my bug us related to this issue -https://github.com/coala/coala/issues/3182

**Comment by NiklasMM** [comment]
Your tests fail, because you have unmet dependencies and therefore incomplete coverage. You can ignore that for now.

First adress the core issues of the PR:

* [ ] Choose a meaningful title for the PR. "jack17529 patch 1" does not tell us anything.
* [ ] Fix the issues pointed out by gitmate-bot. These are coala errors. You can reproduce them by running `coala` in the root directory of the repository. This should always be done before opening a PR.
* [ ] [Rebase](http://www.coala.io/rebase) your branch to have only one commit with a [good commit message](http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html)

**Comment by jack17529** [comment]
@NiklasMM do we need to install csslint by ourself as py.test still gives error on it.

**Comment by NiklasMM** [comment]
Yes, but again: Ignore it. You didn't change anything in the code. It's safe to assume the tests will run just fine in CI. Focus on those issues I listed above.

**Comment by jack17529** [comment]
@NiklasMM coala still gives errors

(env) jack@jack-Latitude-3460:~/coala$ coala
[WARNING][12:36:13] Implicit 'Default' section inheritance is deprecated. It will be removed soon. To silence this warning remove settings in the 'Default' section from your coafile. You can use dots to specify inheritance: the section 'all.python' will inherit all settings from 'all'.
Executing section python...
Executing section autopep8...
Executing section linelength...
Executing section DOCS...
Executing section commit...
Executing section css...
[ERROR][12:36:16] The bear CSSLintBear does not fulfill all requirements. 'csslint' is not installed.
Executing section rst...
Executing section text...
Executing section urllib...
Executing section cli...

**Comment by NiklasMM** [comment]
Ah yes, ok sorry. If you want, you can install [csslint](https://www.npmjs.com/package/csslint), but you'd also need to install node.js. But it's also safe to ignore if you don't change anything in the CSS.

**Comment by jack17529** [comment]
@NiklasMM  done but could not able to check the aims given by you.

File: docs/Developers/Development_Setup.rst [hunk 1]
```diff
@@ -34,6 +34,24 @@ https://github.com/coala/coala-bears
 Installing from Git
 -------------------
 
+NOTE:
+In order to configure the environment according to python 3.
+If you have both python 3 and python 2 installed check this out first.
+
+::
+
+    $ virtualenv venv -p /usr/bin/python3
```
Comment (SanketDG) [inline docs/Developers/Development_Setup.rst]: I did probably do this as: 

```
$ virtualenv venv -p $(which python3)
```

$ virtualenv venv -p `which python3`

Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: @SanketDG so should I change and create a new commit ?

Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: with $ virtualenv venv -p `which python3`

Comment (SanketDG) [inline docs/Developers/Development_Setup.rst]: nah amend the same commit

also I'd go with the first one because it is easy to type

File: docs/Developers/Development_Setup.rst [hunk 2]
```diff
@@ -34,6 +34,24 @@ https://github.com/coala/coala-bears
...
+
+Enter the path where you have installed python 3.
+
+::
+
+    $ source venv/bin/activate
+    (venv)$ python --version
+
+Check your python version and make sure it is python 3.
```
Comment (NiklasMM) [inline docs/Developers/Development_Setup.rst]: We could also mention here that it needs to be python >= 3.4

Comment (adtac) [inline docs/Developers/Development_Setup.rst]: and uppercase P for Python

Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: @NiklasMM  @adtac  ok , sounds like a plan !

File: docs/Developers/Development_Setup.rst (hunk 1 above)
Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: @NiklasMM "I'd put this addition into the "virtualenv" section rather than here"
Can you please elaborate it?

File: docs/Developers/Development_Setup.rst (hunk 2 above)
Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: @NiklasMM "I'd put this addition into the "virtualenv" section rather than here"
Can you please elaborate it?

Comment (NiklasMM) [inline docs/Developers/Development_Setup.rst]: In that same file there's a "virtualenv" section (right at the top). Rather put your addition there as it relates to creating the virtualenv.

**Comment by jack17529** [comment]
oh sorry I do not know how I got them here.

**Comment by NiklasMM** [comment]
Whoops, what happened here? Make sure your branch is properly [rebased](https://coala.io/rebase)

**Comment by jack17529** [comment]
https://github.com/jack17529/coala/commit/a8900a14b51d6c2725727baada850cea84a1ace3

File: docs/Developers/Development_Setup.rst [hunk 3]
```diff
@@ -17,48 +17,57 @@ Here we will be showing how to have a virtualenv using :code:`venv` and
...
+::
+
+        $ cd working_dir
+
+-NOTE:
+ If you have both python 3 and python 2 installed check this out first.
+
+::
+
+	$ virtualenv coala-venv -p $(which python3)
```
Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: Is there a problem with @gitmate-bot  it's showing each twice?

File: docs/Developers/Development_Setup.rst [hunk 4]
```diff
@@ -17,48 +17,57 @@ Here we will be showing how to have a virtualenv using :code:`venv` and
...
+ By using $ which python3
+ We can know the path where python3 is installed.
 
-        $ pip install virtualenv
+::
 
-- Create the virtualenv :
-    ::
+    	$ source coala-venv/bin/activate
+    	(coala-venv)$ python --version
```
Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: @NiklasMM  should I change what gitmate bot said or it's all right.
According to me my this PR was the best it has no issues and should be merged .
Please merge it as it would be my first merged code ,this is critical for submitting my already written proposal.

**Comment by NiklasMM** [comment]
Please fix the issues pointed out by gitmate-bot (you can reproduce them locally by running `coala`). You also did not address the issues pointed out by adtac and myself [here](https://github.com/coala/coala/pull/3922#discussion_r107152052) and [here](https://github.com/coala/coala/pull/3922#pullrequestreview-28105512)

Also refrain from trying to talk us into merging code before it meets even the most basic of quality measures. This behavior is ridiculous and quite frankly offensive and insulting.

**Comment by Adrianzatreanu** [comment]
hey please fix the gitmate issues before asking for review. ugly formatting makes it super hard to review and therefore maintainers will refuse to review your PR

**Comment by jack17529** [comment]
@NiklasMM and @Adrianzatreanu done!

**Comment by sils** [comment]
I don't think GitHub recognizes the "Solves" keyword, please use `Closes` instead as per our docs.

File: docs/Developers/Development_Setup.rst [hunk 5]
```diff
@@ -17,48 +17,57 @@ Here we will be showing how to have a virtualenv using :code:`venv` and
 of the standard library and requires no extra installation. However,
 you can use whichever you find suitable to yourself.
 
-Using venv
-~~~~~~~~~~
+Using virtualenv
```
Comment (NiklasMM) [inline docs/Developers/Development_Setup.rst]: Why did you remove the section on using `venv`? Please restore it. `venv` is not the same as `virtualenv`

File: docs/Developers/Development_Setup.rst [hunk 6]
```diff
@@ -78,8 +87,8 @@ https://github.com/coala/coala-bears
 Installing from Git
 -------------------
 
-We recommend first installing the latest development snapshot of coala's
-master branch from and all of its dependencies with pip using
+-We recommend first installing the latest development snapshot of coala's
+ master branch from and all of its dependencies with pip using
```
Comment (NiklasMM) [inline docs/Developers/Development_Setup.rst]: Unnecessary changes.

File: docs/Developers/Development_Setup.rst [hunk 7]
```diff
@@ -91,28 +100,28 @@ master branch from and all of its dependencies with pip using
     (coala-venv)$ cd coala-bears
     (coala-venv)$ pip3 install -e .
 
-Then you can install a repository-backed version of the repository you would
-like to modify using
+-Then you can install a repository-backed version of the repository you would
+ like to modify using
```
Comment (NiklasMM) [inline docs/Developers/Development_Setup.rst]: Unnecessary changes.

File: docs/Developers/Development_Setup.rst [hunk 8]
```diff
@@ -91,28 +100,28 @@ master branch from and all of its dependencies with pip using
...
     (coala-venv)$ pip3 install -e <path/to/clone>
 
-You will then be able to edit the repository and have the changes take effect
-in your virtualenv immediately. You will also be able to use pip to manage
-your installation of the package should you need to install from a different
-source in the future.
+-You will then be able to edit the repository and have the changes take effect
+ in your virtualenv immediately. You will also be able to use pip to manage
+ your installation of the package should you need to install from a different
+ source in the future.
```
Comment (NiklasMM) [inline docs/Developers/Development_Setup.rst]: Unnecessary changes.

File: docs/Developers/Development_Setup.rst [hunk 9]
```diff
@@ -91,28 +100,28 @@ master branch from and all of its dependencies with pip using
...
 ::
 
     (coala-venv)$ pip3 install -r docs-requirements.txt
 
-Once you have done so, you can build the documentation by entering the docs
-directory and running ``make``. The documentation on the coala website is in
-the ``coala`` (not ``coala-bears``) repository.
+-Once you have done so, you can build the documentation by entering the docs
+ directory and running ``make``. The documentation on the coala website is in
+ the ``coala`` (not ``coala-bears``) repository.
```
Comment (NiklasMM) [inline docs/Developers/Development_Setup.rst]: Unnecessary changes.

File: docs/Developers/Development_Setup.rst (hunk 5 above)
Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: ok sir doing it

File: docs/Developers/Development_Setup.rst (hunk 6 above)
Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: @NiklasMM  If you have rights to my commits editing please edit them , I just only try to make it look the same way , I have no idea how others like it .

Comment (NiklasMM) [inline docs/Developers/Development_Setup.rst]: There should not be any change here at all.

File: docs/Developers/Development_Setup.rst (hunk 9 above)
Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: @NiklasMM will you merge if I changed these , or others will appear saying something else , I am losing hope now.

**Comment by jack17529** [comment]
@NiklasMM  any more changes please tell , otherwise please merge

**Comment by NiklasMM** [comment]
You still have not addressed any of the issues referenced from [this](https://github.com/coala/coala/pull/3922#issuecomment-288528880) comment and your commit is full of unnecessary changes, which I pointed out at least two times now. If you are having problems fixing that, please ask.

The next time you update this, please list the improvements you have made in a comment.

**Comment by jack17529** [comment]
1. Running coala does not give error.
2. Rebased properly.
3. Took the current file and only did a very small change thus removing unnecessary  changes.
4. addition into the "virtualenv" section.
5. Changed "python 3" -> "python3", same for python 2. I think this is more canonical.
6. Changed pip -> pip3 everywhere.
7. Changed python -> Python .
8. Wrote correct commit message. Used Closes instead of solves.

File: docs/Developers/Development_Setup.rst [hunk 10]
```diff
@@ -40,18 +40,29 @@ Using venv
...
 
-- Run coala-venv :
+-NOTE:
+ If you have both Python3 and Python2 installed check this out first.
+
+::
+
+    $ virtualenv coala-venv -p $(which python3)
+
+-This creates a isolated Python3 environment called coala-venv
```
Comment (supergr35) [inline docs/Developers/Development_Setup.rst]: `a isolated` -> `an isolated`

File: docs/Developers/Development_Setup.rst [hunk 11]
```diff
@@ -40,18 +40,29 @@ Using venv
...
+ In your current directory.
+ As coala only works for Python >= 3.4.
```
Comment (supergr35) [inline docs/Developers/Development_Setup.rst]: `In` -> `in`
These two sentences can be combined into one, in my opinion.
```
This creates a isolated Python3 environment called coala-venv In your current directory. As coala only works for Python >= 3.4
```
can become
```
This creates an isolated Python3 environment called coala-venv in your current directory, as coala only works for Python >= 3.4.
```

File: docs/Developers/Development_Setup.rst [hunk 12]
```diff
@@ -40,18 +40,29 @@ Using venv
...
+
+- Re run coala-venv :
```
Comment (supergr35) [inline docs/Developers/Development_Setup.rst]: Why `Re run`? Isn't this the first time entering `coala-venv`?

**Comment by supergr35** [comment]
I feel as if the commit message could be changed to actually identify what this commit is doing. Please address this and make sure it's in the imperative mood.

**Comment by supergr35** [comment]
The commit body can still be changed to describe the change, something more like
```Change all references of pip and python 
to pip3 and python3. Also add a section explaining
what to do with different versions of python installed```

With the correct number of characters and newlines, of course. Also, you have not addressed the changes with capitalization and sentences.

**Comment by NiklasMM** [comment]
It's incredibly frustrating for all of us if you only fix a subset of the issues pointed out each iteration :disappointed:

File: docs/Developers/Development_Setup.rst [hunk 13]
```diff
@@ -40,17 +40,28 @@ Using venv
...
 
+-NOTE:
+ If you have both Python3 and Python2 installed check this out first.
+
+::
+
+    $ virtualenv coala-venv -p $(which python3)
+
+-This creates an isolated Python3 environment called coala-venv
+ In your current directory.
```
Comment (jayvdb) [inline docs/Developers/Development_Setup.rst]: s/In/in/

**Comment by jayvdb** [comment]
Please stop rebasing your PR without fixing the problems that reviewers have told you about.  You are wasting our build cycles.  Please do not ignore reviewers.  It is rude to waste the time of reviewers.

File: docs/Developers/Development_Setup.rst [hunk 14]
```diff
@@ -92,15 +103,15 @@ master branch from and all of its dependencies with pip using
     (coala-venv)$ pip3 install -e .
 
 Then you can install a repository-backed version of the repository you would
-like to modify using
+like to modify using.
```
Comment (jayvdb) [inline docs/Developers/Development_Setup.rst]: revert this incorrect and unrelated change

File: docs/Developers/Development_Setup.rst [hunk 15]
```diff
@@ -75,11 +86,11 @@ https://github.com/coala/coala
 
 https://github.com/coala/coala-bears
 
-Installing from Git
+Installing from git
 -------------------
 
 We recommend first installing the latest development snapshot of coala's
-master branch from and all of its dependencies with pip using
+master branch from and all of its dependencies with pip3 using.
```
Comment (jayvdb) [inline docs/Developers/Development_Setup.rst]: please remove the period at the end.  sentences do not end with `using`

File: docs/Developers/Development_Setup.rst [hunk 16]
```diff
@@ -75,11 +86,11 @@ https://github.com/coala/coala
 
 https://github.com/coala/coala-bears
 
-Installing from Git
+Installing from git
```
Comment (jayvdb) [inline docs/Developers/Development_Setup.rst]: revert this unrelated change

File: docs/Developers/Development_Setup.rst [hunk 17]
```diff
@@ -64,7 +75,7 @@ Repositories
 ------------
 
 If you are interested in contributing to coala, we recommend that you read
-our `newcomers' guide <http://api.coala.io/en/latest/Developers/Newcomers_Guide.html>`__
+our `newcomer's guide <http://api.coala.io/en/latest/Developers/Newcomers_Guide.html>`__
```
Comment (jayvdb) [inline docs/Developers/Development_Setup.rst]: revert this incorrect and unrelated change

File: docs/Developers/Development_Setup.rst [hunk 18]
```diff
@@ -99,8 +110,8 @@ like to modify using
     (coala-venv)$ pip3 install -e <path/to/clone>
 
 You will then be able to edit the repository and have the changes take effect
-in your virtualenv immediately. You will also be able to use pip to manage
-your installation of the package should you need to install from a different
+in your virtualenv immediately. You will also be able to use pip3 to manage
+your installation of the package, you need to install from a different
```
Comment (adtac) [inline docs/Developers/Development_Setup.rst]: This is not a grammatical error. That was a perfectly valid sentence (I even told you so in the chat). Why did you change it?

Comment (adtac) [inline docs/Developers/Development_Setup.rst]: And don't make changes that are not relevant to the issue.

File: docs/Developers/Development_Setup.rst [hunk 19]
```diff
@@ -20,14 +20,14 @@ you can use whichever you find suitable to yourself.
 Using venv
 ~~~~~~~~~~
 
-- Make sure to have Python 3 installed in your local machine.
+- Make sure to have Python3 installed in your local machine.
```
Comment (sils) [inline docs/Developers/Development_Setup.rst]: you're doing this because of https://github.com/coala/coala/issues/3921#issuecomment-286710570 I presume, however in written normal text, not code samples, you usually write `Python 3`, for commands it's `python3`.

File: docs/Developers/Development_Setup.rst [hunk 20]
```diff
@@ -20,14 +20,14 @@ you can use whichever you find suitable to yourself.
...
 
 - Setting up virtualenv with venv :
     ::
 
         $ cd working_dir # move into the dir where you want to create coala-venv
         $ python3 -m venv coala-venv
-        # This creates a isolated python environment called coala-venv
+        # This creates an isolated Python3 environment called coala-venv
```
Comment (sils) [inline docs/Developers/Development_Setup.rst]: same

File: docs/Developers/Development_Setup.rst [hunk 21]
```diff
@@ -40,17 +40,28 @@ Using venv
 Using virtualenv
 ~~~~~~~~~~~~~~~~
 
-- Install virtualenv using pip :
+- Install virtualenv using pip3 :
```
Comment (sils) [inline docs/Developers/Development_Setup.rst]: the tool is pip which should be named here, (the command is pip3 as below)

File: docs/Developers/Development_Setup.rst [hunk 22]
```diff
@@ -40,17 +40,28 @@ Using venv
...
+        $ pip3 install virtualenv
 
 - Create the virtualenv :
     ::
 
         $ cd working_dir # move into the dir where you want to create coala-venv
         $ virtualenv coala-venv
 
+-NOTE:
+ If you have both Python3 and Python2 installed check this out first.
```
Comment (sils) [inline docs/Developers/Development_Setup.rst]: same, please add spaces

**Comment by sils** [comment]
`Add Python 3 check` isn't a very concise description of what you're doing, you're not adding a check, right? How about `Use python3 consistently`?

**Comment by sils** [comment]
About https://github.com/coala/coala/pull/3922#pullrequestreview-28105512 @NiklasMM I disagree with you on that. `python3` is the binary and on https://www.python.org/downloads/ you will find they name it `Python 3`, nobody ever writes `Python3` in english text if it's not about the command.

**Comment by sils** [comment]
@jack17529 can you do a quick rebase? (You know how that works?)

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3922/commits/a8e17dc3f36c0e890f59e91141da155346148631

**Comment by supergr35** [comment]
ack 34f121b

File: docs/Developers/Development_Setup.rst [hunk 23]
```diff
@@ -43,14 +43,25 @@ Using virtualenv
...
+        $ pip3 install virtualenv
 
 - Create the virtualenv :
     ::
 
         $ cd working_dir # move into the dir where you want to create coala-venv
         $ virtualenv coala-venv
 
+-NOTE:
+ If you have both Python 3 and Python 2 installed check this out first.
```
Comment (Mixih) [inline docs/Developers/Development_Setup.rst]: "check this out" -> "try this command"
"check this out" seems to be asking the user to check the output which is not the case here.
Also checkout has been overloaded in open source to also mean switch branch so it would also be confusing.

File: docs/Developers/Development_Setup.rst [hunk 24]
```diff
@@ -43,14 +43,25 @@ Using virtualenv
...
+
+::
+
+    $ virtualenv coala-venv -p $(which python3)
+
+-This creates an isolated Python 3 environment called coala-venv
+ in your current directory.
+ As coala only works for Python >= 3.4.
```
Comment (Mixih) [inline docs/Developers/Development_Setup.rst]: sorry, formatting is strange here. please remove the newline after "directory"  and replace the period with a comma so it reads "in your current directory, as coala only works for Python >= 3.4". Arguably, this text blob could go before the command so the user knows what it does before they see it.

**Comment by Mixih** [comment]
unack 34f121b

File: docs/Developers/Development_Setup.rst (hunk 23 above)
Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: Can I PM you Mixih?

File: docs/Developers/Development_Setup.rst (hunk 24 above)
Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: Sorry,I can't change until you let PM you .

Comment (sils) [inline docs/Developers/Development_Setup.rst]: @jack17529 any reason this can't be discussed here? PMs are only for issues that require privacy and especially not technical issues.

**Comment by jack17529** [comment]
@Mixih  your 2 requested changes are put in my PR carefully.

File: docs/Developers/Development_Setup.rst [hunk 25]
```diff
@@ -43,14 +43,22 @@ Using virtualenv
...
+        $ pip3 install virtualenv
 
 - Create the virtualenv :
     ::
 
         $ cd working_dir # move into the dir where you want to create coala-venv
         $ virtualenv coala-venv
 
+NOTE:
+If you have both Python 3 and Python 2 installed try this command,
```
Comment (Mixih) [inline docs/Developers/Development_Setup.rst]: period at the end of this line please. (modification are advised because this sentence has become too long)

File: docs/Developers/Development_Setup.rst [hunk 26]
```diff
@@ -43,14 +43,22 @@ Using virtualenv
...
+as this creates an isolated Python 3 environment called coala-venv,
```
Comment (Mixih) [inline docs/Developers/Development_Setup.rst]: Please delete `as` and start the sentence on `this`. No comma at end of line please

Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: yes

Comment (jack17529) [inline docs/Developers/Development_Setup.rst]: hope this time it will get merged

**Comment by NiklasMM** [comment]
ack b453a2e

**Comment by jack17529** [comment]
Damn I need a 8k photo of this !!!

**Comment by sils** [comment]
screenshot from 2017-03-26 18-36-15

I can offer 4k ^.

We'll rebase it for you unless you do, probably tomorrow to test our new bot that will automatically rebase things.

**Comment by adtac** [comment]
Sorry for poking into your tabs, but is that relevant XKCD? :D :D

**Comment by sils** [comment]
@adtac course it is! I was searching for your cherry pick bot

**Comment by adtac** [comment]
@cherry-pick-bot

**Comment by adtac** [comment]
@cherry-pick-bot

**Comment by sils** [comment]
merging cherry picked PR :)

**Comment by adtac** [comment]
go home bot, you're drunk
